src/apps/yearn/arbitrum/yearn.v2-vault.token-fetcher.ts

```diff
@@ -1,26 +1,58 @@
 import { Inject } from '@nestjs/common';
 
+import { APP_TOOLKIT, IAppToolkit } from '~app-toolkit/app-toolkit.interface';
 import { Register } from '~app-toolkit/decorators';
-import { CURVE_DEFINITION } from '~apps/curve';
-import { PositionFetcher } from '~position/position-fetcher.interface';
-import { AppTokenPosition } from '~position/position.interface';
+import { DataPropsStageParams, PricePerShareStageParams } from '~position/template/app-token.template.position-fetcher';
 import { Network } from '~types/network.interface';
 
-import { YearnV2VaultTokenHelper } from '../helpers/yearn.v2-vault.token-helper';
+import { YearnContractFactory, YearnVaultV2 } from '../contracts';
+import { YearnVaultTokenDefinitionsResolver } from '../helpers/yearn.vault.token-definitions-resolver';
+import { YearnVaultTokenDataProps, YearnVaultTokenFetcher } from '../helpers/yearn.vault.token-fetcher';
 import { YEARN_DEFINITION } from '../yearn.definition';
 
 const appId = YEARN_DEFINITION.id;
 const groupId = YEARN_DEFINITION.groups.v2Vault.id;
 const network = Network.ARBITRUM_MAINNET;
 
 @Register.TokenPositionFetcher({ appId, groupId, network })
-export class ArbitrumYearnV2VaultTokenFetcher implements PositionFetcher<AppTokenPosition> {
-  constructor(@Inject(YearnV2VaultTokenHelper) private readonly yearnVaultTokenHelper: YearnV2VaultTokenHelper) {}
-
-  async getPositions() {
-    return this.yearnVaultTokenHelper.getTokens({
-      network,
-      dependencies: [{ appId: CURVE_DEFINITION.id, groupIds: [CURVE_DEFINITION.groups.pool.id], network }],
-    });
+export class ArbitrumYearnV2VaultTokenFetcher extends YearnVaultTokenFetcher<YearnVaultV2> {
+  appId = appId;
+  groupId = groupId;
+  network = network;
+
+  vaultType = 'v2' as const;
+  vaultsToIgnore = [];
+
+  constructor(
+    @Inject(YearnContractFactory) private readonly contractFactory: YearnContractFactory,
+    @Inject(YearnVaultTokenDefinitionsResolver)
+    tokenDefinitionsResolver: YearnVaultTokenDefinitionsResolver,
+    @Inject(APP_TOOLKIT) protected readonly appToolkit: IAppToolkit,
+  ) {
+    super(appToolkit, tokenDefinitionsResolver);
+  }
+
+  getContract(address: string): YearnVaultV2 {
+    return this.contractFactory.yearnVaultV2({ network: this.network, address });
+  }
+
+  async getPricePerShare({ contract, appToken }: PricePerShareStageParams<YearnVaultV2>) {
+    const pricePerShareRaw = await contract.pricePerShare().catch(() => 0);
```

Comment:
Use `isMulticallUnderlyingError` in case there's a network error that occurs here